### ISSUE TITLE
Regarding the solution of equation (3) in your paper

### ISSUE THREAD

**Issue by zyang1580** [rutgerswiselab/user-fairness#1]
Hi, thanks for your contributions. When I read your paper, I have a question -- metric M  is used as an extra constraint to find an optimum W, but which data is used to compute metric M? It's validation set?

**Comment by yunqili204** [comment]
Hi, thanks for your question. We use re-ranking algorithm to generate new Top-K recommendation lists based on the original Top-N lists,  which are generated by baseline models on test set. Therefore, the metric M is computed based on those recommendation lists.

**Comment by zyang1580** [comment]
Does it mean that the F1 score(or NDCG@K) is computed on the testing
dataset to find the best W for reranking? If yes, it seems not a common
setting because the ground truth has been utilized to find the best W for
re-ranking. Please be free to point out my misunderstanding.

yunqili204 ***@***.***> 于2021年4月30日周五 上午3:31写道：

**Comment by yunqili204** [comment]
Yes, since the aim of the re-ranking algorithm is to find the fair lists to users, we should re-rank the original Top-N recommendation lists generated by baselines on the test set, and our defined fairness constraint needs the label to compute the evaluation metric. However, we don't think it will be an issue. In real-world system, we can enhance fairness as long as we have some labeled items in the candidate sets when we generate the recommendations, and it can be achieved reasonably by lots of ways depending on specific projects. For example, we can add some interacted items (which can be separated from training set) into the candidate sets, so that we can evaluate fairness of a recommendation list and do the re-ranking, we don’t necessarily need to show such items to users, but use them as a signal to improve fairness. We can also collect user feedbacks online to obtain the label, etc.

**Comment by zyang1580** [comment]
Thanks for your explanation.

yunqili204 ***@***.***> 于2021年5月4日周二 上午8:52写道：
